internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -483,22 +483,30 @@ components:
         - architecture
         - image_type
         - repositories
-        - upload_request
+        - upload_options
       properties:
         architecture:
           type: string
           example: 'x86_64'
         image_type:
-          type: string
-          example: 'ami'
+          $ref: '#/components/schemas/ImageTypes'
         repositories:
           type: array
           items:
             $ref: '#/components/schemas/Repository'
         ostree:
           $ref: '#/components/schemas/OSTree'
-        upload_request:
-          $ref: '#/components/schemas/UploadRequest'
+        upload_options:
+          $ref: '#/components/schemas/UploadOptions'
+    ImageTypes:
+      type: string
+      enum:
+        - aws.ec2
+        - aws.s3
+        - gcp
+        - azure
+        - rhel-edge-commit.aws.s3
+        - rhel-edge-installer.aws.s3
```

Comment:
I'm also unsure if we need to specify `aws.s3` here. Really this should take no options, we just want to upload a blob "somewhere" and get back a URL. Or am I oversimplifying?

Reply:
well I'm not sure, don't we want to have these as bootable images as well at some point? @achilleas-k idk what's best here, remove the `aws.s3` bit?

Reply:
The installer is meant to be downloaded, not to be booted in AWS. We should eventually have edge images in AWS, but they would be their own image type.
Oh, we should call these things `edge-commit` and `edge-installer`, without the `rhel` prefix, no? (Should work for centos too)

Reply:
I agree.  No suffix for blob types, I think.  The upload location shouldn't matter in the end.  The fact that it's s3 doesn't need to be visible and we could (theoretically) change the location at some point and it would (or should) make no difference.

Reply:
https://github.com/osbuild/osbuild-composer/blob/main/internal/distro/rhel84/distro.go#L869 hmm they're explicitly called rhel-edge-* :/

Reply:
let's do that mapping in crc then.